### PR TITLE
refactor: improve the request body for `addSpaceShare`

### PR DESCRIPTION
**Pull request by yu-iskw**
### Description:
I would like to independently declare the type of the request body for `addSpaceShare` as https://github.com/lightdash/lightdash/pull/5954 .

### CODE REVIEW

File: packages/common/src/types/space.ts [hunk 1]
```diff
@@ -22,10 +22,16 @@ export type SpaceSummary = Pick<
     access: string[];
 };
 
-export type CreateSpace = Pick<Space, 'name'> &
-    Partial<Pick<Space, 'isPrivate' | 'access'>>;
+export type CreateSpace = {
+    name: string;
+    isPrivate?: boolean;
+    access: SpaceShare[];
+};
```
Comment (yu-iskw) [inline packages/common/src/types/space.ts:29]: ditto

File: packages/common/src/types/space.ts [hunk 2]
```diff
@@ -22,10 +22,16 @@ export type SpaceSummary = Pick<
...
 
-export type UpdateSpace = Pick<Space, 'name' | 'isPrivate'>;
+export type UpdateSpace = {
+    name: string;
+    isPrivate: boolean;
+};
```
Comment (yu-iskw) [inline packages/common/src/types/space.ts:34]: ditto

File: packages/common/src/types/space.ts [hunk 3]
```diff
@@ -22,10 +22,16 @@ export type SpaceSummary = Pick<
     access: string[];
 };
 
-export type CreateSpace = Pick<Space, 'name'> &
-    Partial<Pick<Space, 'isPrivate' | 'access'>>;
+export type CreateSpace = {
+    name: string;
+    isPrivate?: boolean;
+    access?: SpaceShare[];
+};
```
Comment (yu-iskw) [inline packages/common/src/types/space.ts:29]: ditto

**Review by rephus — APPROVED** [review]
Tested manually on render, it looks good

**Pull request merged**